docs/narr/urldispatch.rst

```diff
@@ -1199,61 +1225,42 @@ route in a set of route predicates:
-
-    config.add_route('y', '/{year}', custom_predicates=(twenty_ten,))
-    config.add_route('ym', '/{year}/{month}', custom_predicates=(twenty_ten,))
-    config.add_route('ymd', '/{year}/{month}/{day}',
-                     custom_predicates=(twenty_ten,))
-
-The above predicate, when added to a number of route configurations ensures
-that the year match argument is '2010' if and only if the route name is 'ymd',
-'ym', or 'y'.
-
-You can also caption the predicates by setting the ``__text__`` attribute. This
-will help you with the ``pviews`` command (see
-:ref:`displaying_application_routes`) and the ``pyramid_debugtoolbar``.
-
-If a predicate is a class, just add ``__text__`` property in a standard manner.
-
-.. code-block:: python
-    :linenos:
-
-    class DummyCustomPredicate1(object):
-        def __init__(self):
-            self.__text__ = 'my custom class predicate'
+    class TwentyTenPredicate:
+        def text(self):
+            return "twenty_ten = True"
 
-    class DummyCustomPredicate2(object):
-        __text__ = 'my custom class predicate'
+        phash = text
 
-If a predicate is a method, you'll need to assign it after method declaration
-(see `PEP 232 <https://www.python.org/dev/peps/pep-0232/>`_).
+        def __call__(self, info, request):
+            if info['route'].name in ('ymd', 'ym', 'y'):
+                return info['match']['year'] == '2010'
 
-.. code-block:: python
-    :linenos:
+    config.add_route_predicate('twenty_ten', TwentyTenPredicate)
+    config.add_route('y', '/{year}', twenty_ten=True)
+    config.add_route('ym', '/{year}/{month}', twenty_ten=True)
+    config.add_route('ymd', '/{year}/{month}/{day}', twenty_ten=True)
 
-    def custom_predicate():
-        pass
-    custom_predicate.__text__ = 'my custom method predicate'
-
-If a predicate is a classmethod, using ``@classmethod`` will not work, but you
-can still easily do it by wrapping it in a classmethod call.
+The above predicate, when added to a number of route configurations ensures
+that the year match argument is ``'2010'`` if and only if the route name is
+``'ymd'``, ``'ym'``, or ``'y'``.
 
-.. code-block:: python
-    :linenos:
+The ``text`` method is a way to caption the predicates.  This
+will help you with the ``pviews`` command (see
+:ref:`displaying_application_routes`) and the
+:term:`pyramid_debugtoolbar`.
 
-    def classmethod_predicate():
-        pass
-    classmethod_predicate.__text__ = 'my classmethod predicate'
-    classmethod_predicate = classmethod(classmethod_predicate)
+The ``phash`` method should return a string that uniquely identifies a
+specific predicate (the name means *predicate hash*).  A good way to do
+that is to use the same argument name and value that in the call to
+``add_route``, like in the examples above.
 
-The same will work with ``staticmethod``, using ``staticmethod`` instead of
-``classmethod``.
+.. XXX add note about predicate matching for boolean custom predicates?
```

Comment:
If I remember right, I had an issue configuring some debug views in my project with a custom predicate `is_authenticated=True/False` (because the `persmission` parameter to `view_config` is not a predicate param), and could not get it to work at some point until I deleted something, making one view config less specific than the other and so matching what I wanted.  It’s not related to *custom* predicates specifically though, so should just remove the line here.